source/Dialog.cpp

```diff
@@ -324,14 +326,34 @@ void Dialog::Init(const string &message, Truncate truncate, bool canCancel, bool
 	this->isMission = isMission;
 	this->canCancel = canCancel;
 	okIsActive = true;
+	isWide = false;
 
 	text.SetAlignment(Alignment::JUSTIFIED);
-	text.SetWrapWidth(WIDTH - 20);
+	text.SetWrapWidth(Width() - 20);
 	text.SetFont(FontSet::Get(14));
 	text.SetTruncate(truncate);
 
 	text.Wrap(message);
 
+	// If the dialog is too tall, then switch to wide mode
+	int maxHeight = Screen::Height() * 3 / 4;
+	if(text.Height() > maxHeight)
+	{
+		isWide = true;
+		// Re-wrap with the new width
+		text.SetWrapWidth(Width() - 20);
+		text.Wrap(message);
+
+		if(text.LongestLineWidth() <= WIDTH)
+		{
+			// Formatted text is long and skinny (ie scan result dialog). Go back
+			// to using the default.
```

Comment:
```suggestion
			// Formatted text is long and skinny (e.g. scan result dialog). Go back
			// to using the default width, since the wide with doesn't help.
```